crates/astria-conductor/src/executor.rs

```diff
@@ -206,24 +241,24 @@ impl<C: ExecutionClient> Executor<C> {
             return Ok(None);
         };
 
-        let prev_block_hash = self.execution_state.clone();
-
+        let prev_execution_block_hash = self.execution_state.clone();
         info!(
             height = block.header.height.value(),
-            parent_block_hash = hex::encode(&prev_block_hash),
+            parent_block_hash = hex::encode(&prev_execution_block_hash),
             "executing block with given parent block",
         );
 
         let txs = txs.into_iter().map(|tx| tx.transaction).collect::<Vec<_>>();
-
         let timestamp = convert_tendermint_to_prost_timestamp(block.header.time)
             .wrap_err("failed parsing str as protobuf timestamp")?;
 
+        // send the data in the sequencer block to the execution client
         let response = self
             .execution_rpc_client
-            .call_do_block(prev_block_hash, txs, Some(timestamp))
+            .call_do_block(prev_execution_block_hash, txs, Some(timestamp))
             .await?;
         self.execution_state = response.block_hash.clone();
+        self.last_safe_block_hash = Hash::try_from(block.block_hash.clone()).unwrap();
```

Comment:
should this be the block's parent hash now, since the just executed block may not be safe? this will probably get less confusing once #193 is done also

Reply:
yes this should be the parent hash

Reply:
similar to my comment above, I had to leave this for now but will reevaluate after #205 is merged

Reply:
made #220 to take a closer look as this